_posts/2020-12-12-cut-scotty-some-slack.md

```diff
@@ -0,0 +1,159 @@
+
+* Ask a more senior coworker about the books, blogs, or Twitter personalities that have had the greatest impact on their thinking.
+* Find an extensibility point in a module you’re working on that’s no longer used—and get rid of it.
+* Fix some inefficient database queries that are just slightly out of the way.
+* Investigate the history of a confusing bit of code, and either add some comments or improve its design.
+* Create a long Slack thread about something you’ve learned—or something that confuses you.
+
+Start claiming your own slack, and keep experimenting with learning and maintenance.
+Given enough time, folks might even start to think of you as a “miracle worker”.
+
+## Appendix
+
+Here’s some stuff that I considered while writing this but couldn’t quite work into the main body.
+
+TV Tropes has an entry for [“Scotty Time”](https://tvtropes.org/pmwiki/pmwiki.php/Main/ScottyTime), the phenomenon in which a quoted estimate is negated by a leader/manager, and the subordinate manages to get the job done in time.
+
+[Parkinson’s Law](https://en.wikipedia.org/wiki/Parkinson%27s_law) states that "work expands so as to fill the time available for its completion".
+A significant danger of following this strategy is that you inflate your estimates but then don’t do anything useful with the extra time.
+It’s critical to maintain discipline!
+
+“A system with no slack is a broken system” is an adage that I picked up while I worked at Faithlife;
+unfortunately, it is impossible to investigate the source of this quotation via Google because of the existence of a certain chat application.
+
+Richard Hamming’s talk [“You and Your Research”](https://www.cs.virginia.edu/~robins/YouAndYourResearch.html) (transcript) contains a number of anecdotes that are relevant to this theme.
+Notable examples are “Great Thoughts Time” and his story of being forced to rush to deliver results on a Friday, even though the person who wanted them had no intention of working over the weekend.
+
+[^1]: *In theory*, this is also a strategy for getting a household out of a paycheck-to-paycheck existence:
+even the smallest amount of budgetary slack can be used to pay down long-term liabilities
+(like college loans)
+faster than necessary.
+Doing this eventually frees up additional resources that can be used to pay down *other* liabilities faster, leading to much more room in the budget without needing to increase one’s revenue.
+In practice, many households struggle to find that initial bit of slack, which is a much bigger (societal) problem.
+
+[^2]: Pun intended, as *Star Trek* is, in fact, a drama. (sorry not sorry)
+
+[^3]: Yes, I got my start *so long ago* that it was not only prior to the demise of Google Reader, but it was even before the *hegemony* of Google Reader, and engineers would share the list of blogs they were following by sending [OPML](https://en.wikipedia.org/wiki/OPML) files around.
+I’m Definitely Not Still Bitter About the Decline of RSS.
```

Comment:
Nobody is more upset about it than Dave Winer (the creator of RSS and OPML), who predicted that Google was going to ruin RSS in like 2005 or 2006. I know, because I very rudely (and naively) mocked him on the internet for his prediction.